### PR TITLE
[Refactor] Meshery Design Page Card Created

### PR DESCRIPTION
**Pull request by subhradwip01**
Signed-off-by: subhradwip01 <subhradwip01@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/FlipCard/FlipCard.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+const getChild=(children,index)=>{
+  if(children.length!==2) throw Error("Flipcard Must Have Two Child");
+  return children[index];
+}
+
+const FlipCard = ({onClick,children}) => {
+  const classes=useStyle()
```
Comment (YashKamboj) [inline ui/components/FlipCard/FlipCard.js:13]: We are not using `useStyles` anymore , please use `styled`

**Review by YashKamboj — CHANGES_REQUESTED** [review]

File: ui/components/FlipCard/FlipCard.style.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+export const FrontCard=styled('div')(()=>({
+  height:"100%",
+  backfaceVisibility:"hidden",
+}))
+
+export const BackCard=styled('div')(()=>({
+  height:"100%",
+  backfaceVisibility:"hidden",
+  transform:"rotateY(180deg)"
+}))
```
Comment (YashKamboj) [inline ui/components/FlipCard/FlipCard.style.js:31]: @subhradwip01 I think you can move this in the `FlipCard.js`
 as it is not a large styling

File: ui/components/DesignComponent/MesheryDesignCard.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          <ActionContainer>
+            <Button
+              variant="conatinedSecondory"
+              onClick={(ev) => genericClickHandler(ev, setSelectedPatterns)}
+              style={{
+                backgroundColor:""
+              }}
+            >
+              <Avatar
+                src="../../public/static/img/patternwhite.svg"
```
Comment (YashKamboj) [inline ui/components/DesignComponent/MesheryDesignCard.js:81]: Just ` /static/img/patternwhite.svg` would work @subhradwip01

**Review by chirag0002 — COMMENTED** [review]
The code of FlipCard.style.js can be merged in FlipCard.js as it is not a large styling file, also it will become easy to understand the styling of the card if it is present in one file.

File: ui/components/FlipCard/FlipCard.style.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      transformStyle:"preserve-3d",
+      transition:"all 0.5s ease",
+      boxShadow : "0 4px 8px 0 rgba(0,0,0,0.2)",
+      backgroundColor : "#fff",
+      cursor : "pointer",
+      transform:`${flipped && 'rotateY(180deg)'}`
+}))
+
+export const FrontCard=styled('div')(()=>({
+  height:"100%",
```
Comment (YashKamboj) [inline ui/components/FlipCard/FlipCard.style.js:23]: @subhradwip01  Remove the height , as this is causing bug in the back side  of the card
Screenshot (189)

File: ui/components/FlipCard/FlipCard.style.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  backfaceVisibility:"hidden",
+}))
+
+export const BackCard=styled('div')(()=>({
+  height:"100%",
```
Comment (YashKamboj) [inline ui/components/FlipCard/FlipCard.style.js:28]: Same, remove height from here too

**Review by YashKamboj — CHANGES_REQUESTED** [review]

File: ui/components/FlipCard/FlipCard.style.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { styled } from "@mui/material/styles";
+
+export const Container=styled('div')(({theme})=>({
+  position:"relative",
+  height:"100%",
+  borderRadius : theme.spacing(1),
+}))
+
+export const InnerCard=styled('div')(({flipped,theme})=>({
```
Comment (YashKamboj) [inline ui/components/FlipCard/FlipCard.style.js:9]: Use `Paper` here , and remove the background, position, border radius and change the name of InnerCard to CardWrapper

File: ui/components/FlipCard/FlipCard.style.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      padding : theme.spacing(2),
+      borderRadius : theme.spacing(1),
+      position:"relative",
+      height:"100%",
```
Comment (YashKamboj) [inline ui/components/FlipCard/FlipCard.style.js:13]: Also remove the height, as when we flip a card the  height of other cards also increase.

File: ui/components/FlipCard/FlipCard.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  return children[index];
+}
+
+const FlipCard = ({onClick,children}) => {
+  const [flipped,setFlipped]=React.useState(false);
+  const Front=getChild(children,0);
+  const Back=getChild(children,1);
+
+  return (
+    <Container>
```
Comment (YashKamboj) [inline ui/components/FlipCard/FlipCard.js:17]: Also remove the `container` as the styles of the container are worthless

File: ui/components/FlipCard/FlipCard.style.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import { styled } from "@mui/material/styles";
+
+export const Container=styled('div')(({theme})=>({
+  position:"relative",
+  height:"100%",
+  borderRadius : theme.spacing(1),
+}))
```
Comment (YashKamboj) [inline ui/components/FlipCard/FlipCard.style.js:7]: Remove this component

**Review by YashKamboj — CHANGES_REQUESTED** [review]

**Comment by subhradwip01** [comment]
Ok...I will change those things according to the review and I will update it by tomorrow morning

File: ui/components/DesignComponent/MesheryDesignCard.js [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+import DoneAllIcon from "@mui/icons-material/DoneAll";
+import RemoveDoneIcon from "@mui/icons-material/RemoveDone";
+import FullscreenIcon from "@mui/icons-material/Fullscreen";
+import FullscreenExitIcon from "@mui/icons-material/FullscreenExit";
+import DeleteIcon from "@mui/icons-material/Delete";
+import SaveIcon from "@mui/icons-material/Save";
+import FlipCard from "../FlipCard/FlipCard"
+const ActionContainer = styled(Box)(({ theme }) => ({
+  display: "flex",
+  gap:"10px",
```
Comment (YashKamboj) [inline ui/components/DesignComponent/MesheryDesignCard.js:21]: use theme.spacing everywhere

File: ui/components/DesignComponent/MesheryDesignCard.js [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+          </div>
+        </div>
+        <div style={{ marginTop: "50px" }}>
+          <ActionContainer>
+            <Button
+              variant="conatinedSecondory"
+              onClick={(ev) => genericClickHandler(ev, setSelectedPatterns)}
+              style={{
+                backgroundColor:""
+              }}
```
Comment (YashKamboj) [inline ui/components/DesignComponent/MesheryDesignCard.js:78]: Remove this

**Review by YashKamboj — CHANGES_REQUESTED** [review]

**Review by YashKamboj — APPROVED** [review]
LGTM!!! , Great Work

**Comment by YashKamboj** [comment]
This PR is good to merge 👍

**Comment by subhradwip01** [comment]
Resolved all requested changes

**Comment by YashKamboj** [comment]
@Chadha93 Can we merge this?

File: ui/components/Patterns/MesheryDesignCard.js [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+  display: "flex",
+  gap:theme.spacing(1),
+  justifyContent:"flex-end"
+}));
+const INITIAL_GRID_SIZE = { xl : 4, md : 6, xs : 12 };
+const MesheryDesignCard = ({
+  name,
+  updated_at,
+  created_at,
+  patterns_file : application_file,
```
Comment (YashKamboj) [inline ui/components/Patterns/MesheryDesignCard.js:29]: remove `application_file` @subhradwip01

Comment (subhradwip01) [reply ui/components/Patterns/MesheryDesignCard.js:29]: ok

Comment (subhradwip01) [reply ui/components/Patterns/MesheryDesignCard.js]: Check once

**Review by YashKamboj — APPROVED** [review]
LGTM!!! Great Work

**Pull request merged**